### ISSUE TITLE
Second zone control missing? 

### ISSUE THREAD

**Issue by nufegiyq** [chamberlain2007/aprilaire-ha#43]
First, let me thank you for taking the time to develop this integration. I generally like my Aprilaire system but have been consistently frustrated with the lack of support and smart functionality. Your efforts have been gaming changing! 
I have a 6000 series 2 zone complete kit. If you don't already know this includes a zone controller, thermostat and senor. From the thermostat I have the ability to switch between controlling my _upstairs zone_ and my _downstairs zone_. Both of these zones have their own individual heating/cooling/auto profiles. I've noticed I can currently only control my _upstairs zone_. Is the inability to control my second _downstairs zone_ something that I have missed during setup or is the feature not currently supported by the integration? 

Thanks again!

**Comment by chamberlain2007** [comment]
This is something that folks have brought up before. The problem is that I unfortunately don't have one of these multi-zone devices so it's not really possible for me to develop against. The documentation is also very vague and doesn't really specify anything about how to do this. So at this point, unless someone who has one is able to contribute to development, I don't think it will be feasible to support this.

**Comment by nufegiyq** [comment]
So at this point, unless someone who has one is able to contribute to development, I don't think it will be feasible to support this.

I would absolutely be willing to contribute in any way that I can. However, be warned, I am quite the novice! ;)

**Comment by nufegiyq** [comment]
> So at this point, unless someone who has one is able to contribute to development, I don't think it will be feasible to support this.

I'm willing to help or contribute in any way that I can. However, be warned, I am a complete novice when it comes to any sort of development.

**Comment by ADeadPixel** [comment]
> > So at this point, unless someone who has one is able to contribute to development, I don't think it will be feasible to support this.
> 
> I'm willing to help or contribute in any way that I can. However, be warned, I am a complete novice when it comes to any sort of development.

I also have a two zone aprilaire house and I started looking into this a ways back and ended up getting distracted by life. If memory serves the main issue is that the second zone does not communicate as expected. There is no ip control for it and when I was digging in the manual/programming guide the second zone is treated oddly/not really mentioned. I couldn't get a solid answer but it looks like it is treated as a writable sensor by the system but that was as far as I got before getting distracted. I want to continue my work on it but with an upcoming wedding it won't be for a while.

**Comment by chamberlain2007** [comment]
So this would really be mostly in the https://github.com/chamberlain2007/pyaprilaire code. You would need to figure out how to communicate with the second zone, implement that in the mock server, expose it via the client, and then in the integration you would then need to expose multiple climate entities. Unfortunately I have no idea how it works as I don't have one and the documentation doesn't really specify anything about it, so I'm of little help.

**Comment by jmhitchcock314** [comment]
As was previously mentioned I’m excited to have this integration. I finally have control over this stupid AprilAire thermostat our builders installed. 

Like the previous poster, I also have a split system. I’m also interested in helping to add the second thermostat. Unfortunately, my python is a little rusty and I’m struggling to work through the code. Is there anything that gives an overview of the files pyaprilaire files?

**Comment by chamberlain2007** [comment]
The pyaprilaire library is documented with pydocs. There is not more documentation than that.

**Comment by repires** [comment]
Hi @chamberlain2007 , thanks again for the great work! Main Thermostat temp and humidifier working great!

I also have a dual zone. Also noticed that Fresh Air System (HRV in my case) is also not available. 

I exported the config file from my Aprilaire Model 6002 to see if it can be of any help.
JSON config file:
https://paste.ofcode.org/md8uVxABz7vvb9uxi6JUwn

Installation manual:
https://www.homecontrols.com/homecontrols/products/pdfs/RP-AprilAire/RP8840-Manual.pdf

I will review the https://github.com/chamberlain2007/pyaprilaire code and get back to you if I come up with any findings.

Cheers

**Comment by chamberlain2007** [comment]
@repires Thanks for this. Unfortunately nothing there really helps me to understand what in the protocol is used for the multiple zones. I still don't see anything in the Aprilaire documentation about that.

**Comment by chamberlain2007** [comment]
I have requested information from Aprilaire support to see if they can provide some feedback on this.

**Comment by chamberlain2007** [comment]
Unfortunately, I contacted Aprilaire and there is no documentation available. Here's the response I got.

---

Hello Matthew,

Unfortunately, our 6000-series of zoning thermostat has been obsoleted,  and no home automation developers ever asked to create drivers for that series of thermostat, thus the information was never added to the developer’s guide and is unavailable.

All our other thermostats outside of the 6000-series, and current home automation capable thermostats, are single-zone thermostats. The thermostats themselves can only control one zone. If using the units in a multi-zone setup, you would need to use a zone panel like the 6403/6404.

---

With that, I really don't think this is going to go anywhere unless someone has a device and can reverse engineer it. It may not even be exposed to begin with.
